docs/azure/sdk/protocol-convenience-methods.md

```diff
@@ -0,0 +1,209 @@
+
+```csharp
+// Create the client
+var safetyClient = new ContentSafetyClient(
+    new Uri("<content-safety-service-uri>"),
+    new DefaultAzureCredential());
+
+// Create the message content
+RequestContent message = RequestContent.Create(new
+    {
+        text = "What is Microsoft Azure?",
+    });
+
+// Call the protocol method
+Response response = safetyClient.AnalyzeText(
+    message, 
+    new RequestContext()
+    {
+         ErrorOptions = ErrorOptions.NoThrow;
+    });
+
+// Display the response data
+using (StreamReader streamReader = new StreamReader(response.ContentStream))
+{
+    Console.WriteLine(streamReader.ReadToEnd());
+}
+```
+
+The preceding code demonstrates the following protocol method patterns:
+
+- Uses the `RequestContent` type as a  to supply data for the request body.
+- Uses the `RequestContext` type to configure request options.
+- Returns data using the `Response` type.
+- Reads the `ContentStream` to access the response data.
+
+#### Work with convenience methods
+
+The following code uses a `ContentSafetyClient` to call the `AnalyzeText` convenience method:
+
+```csharp
+// Create the client
+var safetyClient = new ContentSafetyClient(
+    new Uri("content-safety-service-uri"),
+    new DefaultAzureCredential());
+
+// Call the convenience method
+AnalyzeTextResult result = safetyClient.AnalyzeText("What is Microsoft Azure?");
+
+// Display the response data
+foreach (var item in result.CategoriesAnalysis)
+{
+    Console.Write($"{item.Category}: ");
+    Console.Write(item.Severity);
+    Console.WriteLine();
+}
+```
+
+The preceding code demonstrates the following `Azure.Core` convenience method patterns:
+
+- Uses a standard C# primitive or model type as a parameter.
+- Returns a friendly C# type that represents the result of the operation.
+
+### Libraries that depend on System.ClientModel
+
+Some Azure client libraries depend directly on the `System.ClientModel` library. For example, the [`OpenAI`](https://www.nuget.org/packages/OpenAI/2.0.0-beta.7) library depends on the `System.ClientModel` library and provides a `ChatClient` class that exposes both protocol and convenience methods.
+
+#### Work with protocol methods
+
+The following code uses a `ChatClient` to call the `CompleteChat` protocol method:
+
+```csharp
+// Create the client
+OpenAIClient client = new("your-openai-key");
+ChatClient chatClient = client.GetChatClient("gpt-4");
+
+// Create the request prompt content
+BinaryData prompt = BinaryData.FromBytes("""
+    {  
+        "model": "gpt-4o",
+        "messages": [
+           {
+               "role": "user",
+               "content": "What is Microsoft Azure?."
+           }
+        ]
+    }
+    """u8.ToArray());
+using BinaryContent content = BinaryContent.Create(prompt);
+
+// Send the request
+ClientResult result = chatClient.CompleteChat(
+        content,
+        new RequestOptions()
+        { 
+            ErrorOptions = ClientErrorBehaviors.NoThrow
```

Comment:
Same comment about NoThrow here ... if the service returns a 4xx code, response.Content or JsonDocument.Parse would throw.